doc/source/io.rst

```diff
@@ -440,6 +440,45 @@ individual columns:
     Specifying ``dtype`` with ``engine`` other than 'c' raises a
     ``ValueError``.
 
+.. note::
+
+   Reading in data with mixed dtypes and relying on ``pandas``
+   to infer them is not recommended. In doing so, the parsing engine will
+   loop over all the dtypes, trying to convert them to an actual
+   type; if something breaks during that process, the engine will go to the
+   next ``dtype`` and the data is left modified in place. For example,
```

Comment:
I find this a complex sentence. I think it is easier to explain that the dtypes are inferred (or the data parsed) for parts of the data and not the full dataset at once. And therefore you can end up with a column with mixed dtypes. 

In the alternatives sections, I would also mention the `low_memory` option (since this is what the warning actually suggests). Maybe also give the code example and show that you have a single dtype.